src/simply-rets-api-helper.php

```diff
@@ -1348,17 +1348,33 @@ public static function resultDataColumnMarkup($val, $name, $reverse=false) {
     }
 
 
-    public static function srResidentialResultsGenerator( $response, $settings ) {
-        $br                = "<br>";
+    public static function srResidentialResultsGenerator($request_response, $settings) {
         $cont              = "";
-        $pagination        = $response['pagination'];   // get pagination links out of response
-        $lastUpdate        = $response['lastUpdate'];   // get lastUpdate time out of response
-        $response          = $response['response'];     // get listing data out of response
+        $pagination        = $request_response['pagination'];
+        $lastUpdate        = $request_response['lastUpdate'];
+        $response          = $request_response['response'];
         $MLS_text          = SrUtils::mkMLSText();
         $show_listing_meta = SrUtils::srShowListingMeta();
-        $pag               = SrUtils::buildPaginationLinks( $pagination );
-        $prev_link         = $pag['prev'];
-        $next_link         = $pag['next'];
+
+        /*
+         * Check for an `.error` in the response and return it if it
+         * exists.  Check for NULL is mostly just redundancy.
+        */
+        if(!is_array($response) && property_exists($response, "error")) {
+            $error_message = SrMessages::noResultsMsg($response);
+            return $error_message;
+        }
+
+        if(!is_array($response)) {
+            $response = array($response);
+        }
+
+        /** Build pagination links HTML **/
+        $page_count = count($response);
```

Comment:
This section is the majority of the fix.
Everything else is renaming/reording some things.